pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -385,27 +385,33 @@ func DeleteIPsecEndpoint(remote *net.IPNet) {
 
 func isXfrmPolicyCilium(policy netlink.XfrmPolicy) bool {
 	if policy.Mark == nil {
+		// Check if its our fwd rule, we don't have a mark
```

Comment:
FWIW this commit seems like it could do with a unit test. I don't know how reasonable that is in the immediate term but would be good to note down at least.

Reply:
Do our unit tests run in network namespaces though? Otherwise we will be stomping on the host most likely.

Reply:
We have some tests where we set up the netns as part of the setup/teardown for isolation, such as those in `pkg/datapath/linux/node_linux_test.go`.